Mathlib/CategoryTheory/WithTerminal.lean

```diff
@@ -138,6 +142,115 @@ def map {D : Type*} [Category D] (F : C ⥤ D) : WithTerminal C ⥤ WithTerminal
     | star, star, _ => PUnit.unit
 #align category_theory.with_terminal.map CategoryTheory.WithTerminal.map
 
+/-- A natural isomorphism between the functor `map (𝟭 C)` and `𝟭 (WithTerminal C)`. -/
+@[simps!]
+def mapId (C : Type*) [Category C] : map (𝟭 C) ≅ 𝟭 (WithTerminal C) :=
+  NatIso.ofComponents (fun X => match X with
+    | of x => Iso.refl _
+    | star => Iso.refl _) (by aesop_cat)
+
+/-- A natural isomorphism between the functor `map (F ⋙ G) ` and `map F ⋙ map G `. -/
+@[simps!]
+def mapComp {D E : Type*} [Category D] [Category E] (F : C ⥤ D) (G : D ⥤ E) :
+    map (F ⋙ G) ≅ map F ⋙ map G :=
+  NatIso.ofComponents (fun X => match X with
+    | of x => Iso.refl _
+    | star => Iso.refl _) (by aesop_cat)
+
+/-- From a natrual transformation of functors `C ⥤ D`, the induced natural transformation
+of functors `WithTerminal C ⥤ WithTerminal D`. -/
+@[simps]
+def map₂ {D : Type*} [Category D] {F G : C ⥤ D} (η : F ⟶ G) : map F ⟶ map G where
+  app := fun X => match X with
+    | of x => η.app x
+    | star => 𝟙 star
+  naturality := by
+    intro X Y f
+    match X, Y, f with
+    | of x, of y, f => exact η.naturality f
+    | of x, star, _ => rfl
+    | star, star, _ => rfl
+
+/-- The pseudofunctor from `Cat` to `Cat` defined with `WithTerminal`. -/
+def pseudofunctor : Pseudofunctor Cat Cat where
```

Comment:
You'll want to add `@[simps]` here so that future users of your code also have nice API lemmas.

Reply:
Fixed. Thanks for pointing this out.